lib/ansible/playbook/role/__init__.py

```diff
@@ -274,22 +275,49 @@ def _load_role_data(self, role_include, parent_role=None):
                 raise AnsibleParserError("The handlers/main.yml file for role '%s' must contain a list of tasks" % self._role_name,
                                          obj=handler_data, orig_exc=e)
 
-    def _prepend_validation_task(self, task_data):
+    def _get_role_argspecs(self):
+        '''Get the role argument spec data.
+
+        Role arg specs can be in one of two files in the role meta subdir: argument_specs.yml
+        or main.yml. The former has precendence over the latter. Data is not combined
+        between the files.
+
+        :returns: A dict of all data under the top-level 'argument_specs' YAML key
+            in the argument spec file. An empty dict is returned if there is no
+            argspec data.
+        '''
+        extensions = ('.yml', '.yaml')
```

Comment:
I'm wondering if it would be better to store this list centrally somewhere. We have it defined in: 

https://github.com/ansible/ansible/blob/aea0c4dd1bf04cfff12b6a1e01d99653787048c3/lib/ansible/utils/collection_loader/_collection_finder.py#L73

But that's probably not where it should be referenced from. Maybe `lib/ansible/constants.py`?

Reply:
we already have this covered, use the existing `C.YAML_FILENAME_EXTENSIONS `

Reply:
Awesome! I didn't know that.